enumerable.rb

```diff
@@ -1,50 +1,88 @@
 module Enumerable
-    def my_each
-        if block_given?
-            for x in self do
-                yield (x)
-            end
-        end
-        return self
-    end
-    
-    def my_each_with_index
-        if block_given?
-            for x in self do
-                yield (x)                
-            end
-            for y in self do
-                y.index(self[x]).to_i
-            end
-        end
-        return self
-    end
-    ['a', 'b', 'c'].my_each_with_index { |el, i| puts i }
-    def my_select
-
+  def my_each
+    if block_given?
+      each do |x|
+        yield x
+      end
     end
+    self
+  end
 
-    def my_all?
-
+  def my_each_with_index
+    if block_given?
+      idx = 0
+      each do |x|
+        yield(x, idx)
+        idx += 1
+      end
     end
+    self
+  end
 
-    def my_any?
+  def my_select
+    return unless block_given?
 
+    array = []
+    each do |x|
+      array.push(self[x]) if yield(self[x]) == true
     end
+    array
+  end
 
-    def my_none?
+  def my_all?
+    my_each { |x| return false if yield(x) == false }
+    true
+  end
 
-    end
+  def my_any?
+    my_each { |x| return true if yield(x) == true }
+    false
+  end
 
-    def my_count
+  def my_none?
+    my_each { |x| return false if yield(x) == true }
+    true
+  end
 
+  def my_count(par = nil)
+    number = 0
+    if block_given?
+      my_each do |x|
+        number += 1 if yield(x) == true
+      end
+    elsif !par.nil?
+      my_each do |x|
+        number += 1 if x == par
+      end
+    else
+      number = length
     end
+    number
+  end
 
-    def my_map
-
+  def my_map()
+    arr = []
+    if block_given?
+      my_each do |x|
+        arr.push(yield(x))
+      end
     end
+    arr
+  end
```

Comment:
#### my_map 

- [ ] returns an Enumerator if no block is given

> [2,4,7,11].my_map should be equal to [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
`my_proc = Proc.new {|num| num > 10 }`

Result
> puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  # true true false false